types/avs.go

```diff
@@ -8,6 +8,9 @@ import (
 
 type TaskIndex = uint32
 type TaskResponseDigest = Bytes32
+type TaskResponse = interface{}
+
+type TaskResponseHashFunction func(taskResponse TaskResponse) TaskResponseDigest
```

Comment:
I was originally thinking we'd take/store a SerializedTaskResponse of type `[]byte`, forcing the service user to serialize his task before sending it to us (and he can deserialize when we send him back the response). wydt about doing that instead? any reason why any is needed (I tend to try to stay away from `any` as much as possible).

Also should we use the standard golang Hash interface https://pkg.go.dev/hash#Hash instead of TaskResponseHashFunction? I've personally never used it, but if its already the standard, then we should probably adopt it?

Reply:
Thinking about this more. Actually your `any` is nice because then user can just pass whatever struct they use internally.
But then the TaskResponseHashFunction has to both serialize + hash. Actually don't hate this.